lib/hub/action_request.js

```diff
@@ -122,55 +122,41 @@ class ActionRequest {
-                        .on("error", (err) => {
-                        if (hasResolved && err.code === "ECONNRESET") {
-                            winston.info(`[stream] ignoring ECONNRESET that occured after streaming finished`, this.logInfo);
-                        }
-                        else {
-                            winston.error(`[stream] request stream error`, Object.assign(Object.assign({}, this.logInfo), { error: err.message, stack: err.stack }));
-                            reject(err);
-                        }
-                    })
-                        .on("finish", () => {
-                        winston.info(`[stream] streaming via download url finished`, this.logInfo);
-                    })
-                        .on("socket", (socket) => {
-                        winston.info(`[stream] setting keepalive on socket`, this.logInfo);
-                        socket.setKeepAlive(true);
-                    })
-                        .on("abort", () => {
-                        winston.info(`[stream] streaming via download url aborted`, this.logInfo);
-                    })
-                        .on("response", () => {
-                        winston.info(`[stream] got response from download url`, this.logInfo);
-                    })
-                        .on("close", () => {
-                        winston.info(`[stream] request stream closed`, this.logInfo);
-                    })
-                        .pipe(stream)
-                        .on("error", (err) => {
-                        winston.error(`[stream] PassThrough stream error`, Object.assign({}, this.logInfo));
-                        reject(err);
-                    })
-                        .on("finish", () => {
-                        winston.info(`[stream] PassThrough stream finished`, this.logInfo);
-                        resolve();
-                        hasResolved = true;
-                    })
-                        .on("close", () => {
-                        winston.info(`[stream] PassThrough stream closed`, this.logInfo);
-                    });
+                    try {
+                        (0, node_fetch_1.default)(url).then((response) => {
+                            if (response.ok) {
+                                response.body.pipe(stream)
+                                    .on("error", (err) => {
+                                    winston.error(`[stream] PassThrough stream error`, this.logInfo);
+                                    reject(err);
+                                })
+                                    .on("finish", () => {
+                                    winston.info(`[stream] PassThrough stream finished`, this.logInfo);
+                                    resolve();
+                                })
+                                    .on("close", () => {
+                                    winston.info(`[stream] PassThrough stream closed`, this.logInfo);
```

Comment:
nit: our winston logging messages all have different formats. This one is quite terse compared to the chattier messages below.

Reply:
I was generally trying to keep our common ones still terse, the below error messages will get passed back to looker as well as getting logged so I tried to make it more presentable